routes/api/v1/forecast.js

```diff
@@ -15,41 +16,26 @@ router.get('/', (request, response) => {
     // QUERY THE DATABASE TO CHECK FOR A USER WITH THE PASSED IN API KEY
     database('users').where('api_key', request.body.api_key).first()
       .then((user) => {
-
         if (user) {
-        // CHANGE THE BELOW FUNCTIONS TO HELPER FUNCTIONS IF REFACTORING
-        // START OF THE GOOGLE API SERVICE FUNCTION
-        const location  = request.query.location;
-        const googleApiKey = process.env.GOOGLE_API_KEY;
-
-        const googleApiUrl =`https://maps.googleapis.com/maps/api/geocode/json?address=${location}&key=${googleApiKey}`;
-        fetch(googleApiUrl, { method: 'GET'})
-          .then((response) => {
-             return response.json();
-        })
-        .then((json) => {
-          const googleGeocodeResponse = json;
-          const latAndLng = googleGeocodeResponse.results[0].geometry.location;
-          const joinedLatAndLng = `${latAndLng.lat},${latAndLng.lng}`;
-
-          // START OF DARKSKY API FETCHING
-          const latAndLngFromGoogle = joinedLatAndLng;
+        // GOOGLE HELPER METHOD TO GET THE LAT AND LONG OF THE LOCATION
+        const googleResults = getLatAndLongFromGoogle(request.query.location);
+        googleResults.then( (latAndLongGoogleResults) => {
+        // DARK SKY METHOD GOES HERE
           const darkSkyApiKey = process.env.DARKSKY_API_KEY;
-          const darkSkyUrl = `https://api.darksky.net/forecast/${darkSkyApiKey}/${latAndLngFromGoogle}?exclude=minutely,flags&units=us`;
+          const darkSkyUrl = `https://api.darksky.net/forecast/${darkSkyApiKey}/${latAndLongGoogleResults}?exclude=minutely,flags&units=us`;
```

Comment:
Maybe a next step would be to abstract out your dark sky service?

Reply:
That is ineed the next step!